src/Button.py

```diff
@@ -7,40 +7,70 @@ class Button:
 
     def __init__(self,
                  text,  # The text to display (can be None)
-                 text_colour,  # The colour of the text (can be None)
+                 text_colour,  # The colour of the text/polygons (can be None)
                  background_colour,  # The colour of the text (can be None)
+                 shape,  # list of vectors for polygon (can be None)                 
                  screen_location,  # The position on the screen
                  size):  # The size of the Button
         """Create a Button."""
         self.text = text
         self.text_colour = text_colour
         self.background_colour = background_colour
+        self.shape = shape
         self.screen_location = screen_location
         self.size = size
         self.rect = pygame.Rect(screen_location, size)
         self.font = pygame.font.SysFont("comicsansms", 22)
         self.swapped = False  # Keeps track of wether a Button is swapped
 
+
+    def get_vertex_list(self, array, centerx, centery):
```

Comment:
Does this takes the list of points and makes them centered around `centerx` and `centery`, correct?

Could this be called in `__init__` so that the  display method does minimal processing?